### PR TITLE
Added support for ModelDefinitions.views field

### PR DESCRIPTION
**Pull request by pawartur**
## Description

This will unblock us to enable devs to mark some fields as referencing document controller their GraphQL SDL definitions of models.

The `views` field should be:
```
interface ModelDefinition {
  ...
  views?: ModelViewsDefinition
}
```
, where
```
type ModelViewsDefinition = Record<string, ModelViewDefinition>
```
and
```
ModelViewDefinition = { type: 'documentAccount' }
```

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [X] Added unit tests for the new `ViewsValidation` interface
- [X] Updated tests for `ModelStreamHandler`

## PR checklist

Before submitting this PR, please make sure:

- [x] I have tagged the relevant reviewers and interested parties
- [x] I have updated the READMEs of affected packages
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts
Comment (stbrody) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:104]: I thought these view properties wouldn't show up in the schema itself, but it just metadata for the client library to expose the stream metadata with a given name?  @PaulLeCam

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts
Comment (stbrody) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:339]:
```suggestion
  it('supports view properties in genesis commit', async () => {
```

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts
Comment (pawartur) [reply packages/stream-model-handler/src/__tests__/model-handler.test.ts:104]: Yes! That's right, sorry for the mistake.

**Review by PaulLeCam — APPROVED** [review]
Looks great, thanks!
I'm not the most familiar with this codebase so better wait for approval from other reviewers, but it looks all good to me.

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts
Comment (ukstv) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:97]: Based on what I see, you could do something like
```typescript
{
  ...FINAL_CONTENT,
  views: {
    owner: {
      type: 'documentAccount'
    }
  }
}
```

File: packages/stream-model-handler/src/model-handler.ts
Comment (ukstv) [inline packages/stream-model-handler/src/model-handler.ts:129]:
```suggestion
    if (state.content.schema) {
```

IMO this should work too, and is a bit more idiomatic.

File: packages/stream-model-handler/src/model-handler.ts
Comment (ukstv) [inline packages/stream-model-handler/src/model-handler.ts:131]:
```suggestion
      if (state.content.views) {
```

File: packages/stream-model-handler/src/views-utils.ts
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:12]:
```suggestion
  ): void {
```

File: packages/stream-model-handler/src/views-utils.ts
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:13]:
```suggestion
    Object.entries(views).forEach(([key, value]) => {
```

you do not really return anything here

File: packages/stream-model-handler/src/views-utils.ts
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:14]: Suggested micro-optimisation. Replace `SUPPORTED_VIEW_TYPES` array with an object like
```
{
documentAccount: true
}
```

Then this line would be O(1): `SUPPORTED_VIEW_TYPES[value.type]`.

**Review by ukstv — APPROVED** [review]
few beautifications here and there are possible, but otherwise LGTM

🏇

**Review by stbrody — APPROVED** [review]
two small nitpicks, but overall looks good

**Pull request merged**
